src/machine/board_rpi3.go

```diff
@@ -0,0 +1,182 @@
+// +build rpi3
+
+package machine
+
+import (
+	"device/arm"
+)
+
+//specific to raspberry pi 3
+var MemoryMappedIO = uintptr(0x3F000000)
+
+
+// XXX initialized in preinit so we avoid interp.Unsupported
+var Aux *AuxPeripheralsRegisterMap //= (*AuxPeripheralsRegisterMap)(unsafe.Pointer(MemoryMappedIO + 0x00215000))
+var GPIO *GPIORegisterMap //= (*GPIORegisterMap)(unsafe.Pointer(MemoryMappedIO + 0x00200000))
+var SysTimer *SysTimerRegisterMap //= (*SysTimerRegisterMap)(unsafe.Pointer(MemoryMappedIO + 0x3000))
```

Comment:
I think that using volatile registers and making `MemoryMappedIO` a constant should fix the issue and let you avoid the preinit.

Reply:
Yep, it did.